legend-engine-executionPlan-execution/src/test/java/org/finos/legend/engine/plan/execution/nodes/helpers/freemarker/TestFreeMarkerExecutor.java

```diff
@@ -47,4 +56,29 @@ public static String collectionSizeTemplate()
                 "<#return collection?size> " +
                 "</#function>";
     }
+
+    @Test
+    public void testQueryTagReferralHeaderPlaceHolder()
```

Comment:
This class does not seem a good place for this test.  This should be part of plan execution.  
The issue here is that you are not really testing what happens in production, and actually you end duplicating that flow.